feat(curriculum): add fruit search useEffect workshop (#59544)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/67ec4a95b609c071f7747be0.md

---
id: 67ec4a95b609c071f7747be0
title: Step 3
challengeType: 0
dashedName: step-3
---

# --description--

> Inside the `form` element, add a `label` element with a `for` attribute pointing to `search-input`.

Reviewer: ```suggestion
Inside the `form` element, add a `label` element. Your `label` element should have the text `Search for fruits:` and an `htmlFor` attribute with the value of `"search-input"`.
```

# --hints--

You should have a `label` element inside the `form` element.

```js
const form = document.querySelector("form");
assert.exists(form);

const label = form.querySelector("label");
assert.exists(label);
assert.strictEqual(label.tagName, "LABEL");
```

> The `label` element should have a `for` attribute pointing to the search input
> 
> ```js
> const form = document.querySelector("form");
> assert.exists(form);
> 
> const label = form.querySelector("label");
> assert.exists(label);
> assert.strictEqual(label.tagName, "LABEL");
> assert.strictEqual(label.getAttribute("for"), "search-input");
> ```

Reviewer: the hint needs to be for the `htmlFor` attribute since this is JSX. not regular HTML. Also the hint needs to mention the correct value. 

this should work for the expected outcome

```suggestion
Your `label` element should have an `htmlFor` attribute with the value of `"search-input"`.

```js
const label = document.querySelector("form label");
assert.exists(label);

const key = Object.keys(label).find(key => key.startsWith("__reactProps"));
const reactInstance = label[key]
assert.exists(reactInstance);

assert.equal(reactInstance.htmlFor, "search-input");
```
````

---
id: 67ec4a95b609c071f7747be0
title: Step 3
challengeType: 0
dashedName: step-3
---

# --description--

Inside the `form` element, add a `label` element. Your `label` element should have the text `Search for fruits:` and an `htmlFor` attribute with the value of `"search-input"`.

# --hints--

You should have a `label` element inside the `form` element.

```js
const form = document.querySelector("form");
assert.exists(form);

const label = form.querySelector("label");
assert.exists(label);
assert.strictEqual(label.tagName, "LABEL");
```

> The `label` element should have an `htmlFor` attribute pointing to the search input

Reviewer: ```suggestion
The `label` element should have an `htmlFor` attribute with the value of `"search-input"`.
```

```js
const label = document.querySelector("form label");
assert.exists(label);

const key = Object.keys(label).find(key => key.startsWith("__reactProps"));
const reactInstance = label[key]
assert.exists(reactInstance);

assert.equal(reactInstance.htmlFor, "search-input");
```

> 

Reviewer: there is no test for the label text.
that needs to be added.
